server/app/routes.py

```diff
@@ -112,3 +111,158 @@ def delete_user(user_id):
+
+@app.route('/api/stores/<int:store_id>', methods=['PUT'])
+def update_store(store_id):
+    store = Store.query.get(store_id)
+    if not store:
+        return jsonify({
+            'status': 'Failed',
+            'message': 'Store not found',
+            'data': None
+        }), 404
+    
+    data = request.get_json()
+    store.store_name = data.get('store_name', store.store_name)
+    store.location = data.get('location', store.location)
+    db.session.commit()
+    return jsonify(store.to_dict()), 200
+
+@app.route('/api/stores/<int:store_id>', methods=['DELETE'])
+def delete_store(store_id):
+    store = Store.query.get(store_id)
+    if not store:
+        return jsonify({
+            'status': 'Failed',
+            'message': 'Store not found',
+            'data': None
+        }), 404
+    
+    db.session.delete(store)
+    db.session.commit()
+    
+    return jsonify({
+        "status": "Success",
+        "message": "Store deleted successfully.",
+        "data": None
+    }), 200
+
+# Routes for products
+@app.route('/api/products', methods=['GET'])
+def get_products():
+    products = Product.query.all()
+    products_data = [product.to_dict() for product in products]
+    return jsonify(products_data), 200
+
+@app.route('/api/products/<int:product_id>', methods=['GET'])
+def get_product(product_id):
+    product = Product.query.get(product_id)
+    if not product:
+        return jsonify({
+            'status': 'Failed',
+            'message': 'Product not found',
+            'data': None
+        }), 404
+    
+    return jsonify(product.to_dict()), 200
+
+@app.route('/api/products', methods=['POST'])
+def create_product():
+    data = request.get_json()
+    if 'product_name' not in data or 'buying_price' not in data or 'selling_price' not in data or 'store_id' not in data:
+        return jsonify({
+            'status': 'Failed',
+            'message': 'product_name, buying_price, selling_price, and store_id are required',
+            'data': None
+        }), 400
+    
+    new_product = Product(
+        product_name=data['product_name'],
+        buying_price=data['buying_price'],
+        selling_price=data['selling_price'],
+        store_id=data['store_id']
+    )
+    db.session.add(new_product)
+    db.session.commit()
+    return jsonify(new_product.to_dict()), 201
+
+@app.route('/api/products/<int:product_id>', methods=['PUT'])
+def update_product(product_id):
+    product = Product.query.get(product_id)
+    if not product:
+        return jsonify({
+            'status': 'Failed',
+            'message': 'Product not found',
+            'data': None
+        }), 404
+    
+    data = request.get_json()
+    product.product_name = data.get('product_name', product.product_name)
+    product.buying_price = data.get('buying_price', product.buying_price)
+    product.selling_price = data.get('selling_price', product.selling_price)
+    db.session.commit()
+    return jsonify(product.to_dict()), 200
```

Comment:
Keep the response format consistent so that it maintains a consistent structure.
Use this as the reference:
```python
return jsonify({
            "status": "Failed",
            "message": "Email and user_id are required.",
            "data": None
        }), 400
```